akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -78,6 +77,22 @@ object Sink {
     new Sink(scaladsl.Sink.asPublisher(fanout == AsPublisher.WITH_FANOUT))
 
   /**
+   * A `Sink` that materializes into a [[org.reactivestreams.Publisher]] with additional cancel method.
+   * The difference from [[Sink.asPublisher]] is that sink is still doing backpressure without publishers
+   * connected and does not finalize stream when last publisher cancels. It completes state when cancel
+   * is called or complete came from upsream.
+   *
+   * The materialized `Publisher` will support multiple `Subscriber`s and
+   * the size of the `inputBuffer` configured for this stage becomes the maximum number of elements that
+   * the fastest [[org.reactivestreams.Subscriber]] can be ahead of the slowest one before slowing
+   * the processing down due to back pressure.
+   *
+   * @see [[#asPublisher]]
+   */
+  def asPublisher[T](fanout: Boolean, finalizeOnLastSubscriptionCompletion: Boolean): Sink[T, Publisher[T]] =
```

Comment:
Actually it was discussion about deprecation legacy asPublisher method.
and about method signature before https://github.com/akka/akka/pull/20422#discussion_r61790735. At first it was three boolean arguments.

But I like the idea with asDurablePublisher as we don't need to deprecate anything and API will look more concise

Other opinions about method names?